WiserTaskScheduler/WiserTaskScheduler/Modules/ServerMonitors/Services/ServerMonitorsService.cs

```diff
@@ -0,0 +1,433 @@
+        /// <returns></returns>
+        private async Task GetHardDriveSpaceAsync(ServerMonitorModel monitorItem, int threshold, CommunicationsService gclCommunicationsService, string configurationServiceName, string driveName)
+        {
+            DriveInfo drive = new DriveInfo(driveName);
+
+            if (!emailDrivesSent.ContainsKey(drive.Name))
+            {
+                emailDrivesSent[drive.Name] = false;
+            }
+
+            //Calculate the percentage of free space availible and see if it matches with the given threshold.
+            double freeSpace = drive.TotalFreeSpace;
+            double fullSpace = drive.TotalSize;
+            double percentage = freeSpace / fullSpace * 100;
+
+            //Set the email settings correctly
+            receiver = monitorItem.EmailAddressForWarning;
+            subject = $"Low space on Drive {drive.Name}";
+            body = $"Drive {drive.Name} only has {percentage}% space left, this is below the threshold of {threshold}";
+
+            await logService.LogInformation(logger, LogScopes.RunStartAndStop, monitorItem.LogSettings, $"Drive {drive} has {drive.TotalFreeSpace}Bytes of free space", configurationServiceName, monitorItem.TimeId, monitorItem.Order);
+
+            //Check if the threshold is higher then the free space available.
+            if (percentage < threshold)
+            {
+                await logService.LogInformation(logger, LogScopes.RunStartAndStop, monitorItem.LogSettings, $"Drive {drive.Name} only has {percentage}% space left, this is below the threshold of {threshold}", configurationServiceName, monitorItem.TimeId, monitorItem.Order);
+
+                //Only send an email if the Drive threshold hasn't already been reached.
+                if (!emailDrivesSent[drive.Name])
+                {
+                    emailDrivesSent[drive.Name] = true;
+                    await gclCommunicationsService.SendEmailAsync(receiver, subject, body);
+                }
+            }
+            else
+            {
+                emailDrivesSent[drive.Name] = false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the available RAM space and caclulates if it needs to send an email.
+        /// </summary>
+        /// <param name="monitorItem">The information for the monitor action.</param>
+        /// <param name="threshold">The threshold to check if an email needs to be send.</param>
+        /// <param name="gclCommunicationsService">The name of the service in the configuration, used for logging.</param>
+        /// <param name="configurationServiceName">The communications service from the GCL to actually send out the email.</param>
+        /// <returns></returns>
+        private async Task GetRAMSpaceAsync(ServerMonitorModel monitorItem, int threshold, CommunicationsService gclCommunicationsService, string configurationServiceName)
+        {
+            double ramValue = ramCounter.NextValue();
```

Comment:
Graag hier ook `var` gebruiken, zodat het consistent is met de rest van de code.